File: problems/1588/solution.cpp
class Solution {
public:
    int sumOddLengthSubarrays(vector<int>& arr) {
        int ans = 0;
        
        for (int left = 1, right = arr.size(); left - 1 < arr.size(); left++, right--) {
            ans += (left * right + 1) / 2 * arr[left - 1];
        }
        
        return ans;
    }
};

File: problems/1588/solution.go
package main
func sumOddLengthSubarrays(arr []int) int {
    ans := 0
    
    for left, right := 1, len(arr); left - 1 < len(arr); left, right = left + 1, right - 1 {
        ans += (left * right + 1) / 2 * arr[left - 1]
    }
    
    return ans
}